### ISSUE TITLE
[Bug report] [iOS] media downloaded from web are sometimes not displayed + wrong thumbs for contents

### ISSUE THREAD

**Issue by iosephmagno** [fluttercandies/flutter_photo_manager#935]
### Version

2.1.4

### Platforms

iOS

### Device Model

Iphone 13 pro ios 16.1

### flutter info

```sh
[✓] Flutter (Channel master, 3.11.0-2.0.pre.31, on macOS 12.6.5 21G531 darwin-arm64, locale en-IT)
[!] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
   Run `path/to/sdkmanager --install "cmdline-tools;latest"`
      See https://developer.android.com/studio/command-line for more details.
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/macos#android-setup for more details.
[✓] Xcode - develop for iOS and macOS (Xcode 14.0.1)
[✓] Android Studio (version 2022.2)
[✓] Network resources
```

### How to reproduce?

You can test the issue running wechat_asset_picker on iOS. 
- Sometimes assets downloaded from web are not displayed in the gallery.
- Sometime assets downloaded from web are displayed in the gallery initially but won't be displayed later on on subsequent launches.
- Sometimes thumbs are not the one related to the asset. This means that if user taps a thumb, a different asset is loaded.

I couldn't test this issue on v2.6.0 because of https://github.com/fluttercandies/flutter_photo_manager/issues/921 that prevent me from fetching albums on iOS.

### Logs

```sh
Console throws no errors for above issues.
```

**Comment by AlexV525** [comment]
We generally do not support a previous version. Please let me know if this is reproducible with the latest released version of the plugin. Also, there are already related solutions for your description which have already been explained in previous issues.

**Comment by iosephmagno** [comment]
Tested on 2.6.0 and issue is not reproducible

**Comment by iosephmagno** [comment]
Unfortunately, this issue occurred again. I have used wechat picker demo app as it is (with latest version of photomanager) to make sure that it happens also there. So basically images downloaded from web disappear from Recents album after a while. They are displayed and also positioned correctly inside the grid as soon as downloaded, but they disappear after some app launches, and grid is no longer a correct replica of the phone photo gallery. No errors from console.

**Comment by delfme** [comment]
> We generally do not support a previous version. Please let me know if this is reproducible with the latest released version of the plugin. Also, there are already related solutions for your description which have already been explained in previous issues.

Hi @AlexV525 can I know why this is invalid/outdated? I tested extensively the issue and I can comfirm that images downloaded from web disappear from “Recents” album over time. Tested also with your wechat picker  to make sure it was not my code. Can yoy please have a second look at it? It is pretty severe issue. Peopke download many images from the web and over time the gallery from photomanager will miss many files they have on the phone native gallery.

**Comment by delfme** [comment]
Random thought: since images are downloaded and placed perfectly initially, might it be that this is an issue with the sorting date? I have thousands of images and cannot check if images are still available somewhere in the Recents album, but if I could force the sort by date, I’d love to test it.

**Comment by AlexV525** [comment]
Try `ignoreSize`.

**Comment by delfme** [comment]
```
Future<Map<AssetPathEntity, Uint8List?>> loadAlbums() async {
  final FilterOptionGroup _filterOption = FilterOptionGroup(
    imageOption: const FilterOption(
      sizeConstraint: SizeConstraint(ignoreSize: true),
    ),
  );
```

Already doing it. Is it a way to force sorting? 

Would appreciate if you could check bug on wechat picker, if you get to fix it there I could easily adjust. But I think it is more likely an iOS bug in photomanager.

**Comment by delfme** [comment]
@AlexV525 can we reopen this one to avoid it go lost? I confirm issue occurs even with `ignoreSize` and also with wechat picker demo.

How to reproduce it:
- Download a few images from web
- Use wechat picker demo several times during subsequent days
- At some point the downloaded images will disappear from Recents album and gallery is no longer a faithful copy of the device native gallery

**Comment by AlexV525** [comment]
Could you use the example of the plugin to determine which plugin introduce the behavior?

**Comment by delfme** [comment]
Will try it. Are we sure the issue wasnt alredy there before? Coz it is random and difficult to see at first glance

**Comment by delfme** [comment]
@AlexV525 cannot debug this, nor find a working version. Maybe bug was always there. 

If that can help, I downloaded a few pics during latest 3 days. Yesterday some pics disappeared and I cannot find out a pattern. For example, one pic did disappear but the other one downloaded the exact same day didnt. The format is the same (jpeg) and the size is medium/small. I downloaded them from unsplash https://unsplash.com
